syft/pkg/cataloger/golang/licenses.go

```diff
@@ -0,0 +1,112 @@
+package golang
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/anchore/syft/internal/licenses"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/source"
+)
+
+type goLicenses struct {
+	searchLocalGoModForLicenses bool
+	localGoModResolver          source.FileResolver
+}
+
+func newGoLicenses(searchLocalGoModForLicenses bool) goLicenses {
+	return goLicenses{
+		searchLocalGoModForLicenses: searchLocalGoModForLicenses,
+		localGoModResolver:          deferredResolver,
+	}
+}
+
+// this needs to be shared between GoMod & GoBinary so it's only scanned once
+var deferredResolver = deferredResolverForLocalGoMod()
+
+func deferredResolverForLocalGoMod() source.FileResolver {
+	return source.NewDeferredResolverFromSource(func() (source.Source, error) {
+		goPath := os.Getenv("GOPATH")
+
+		if goPath == "" {
+			homeDir, err := os.UserHomeDir()
+			if err != nil {
+				log.Debug("unable to determine user home dir: %v", err)
+			}
+			goPath = path.Join(homeDir, "go")
+		}
+
+		return source.NewFromDirectory(path.Join(goPath, "pkg", "mod"))
+	})
+}
+
+func (c *goLicenses) getLicenses(resolver source.FileResolver, moduleName, moduleVersion string) (licenses []string, err error) {
+	nameParts := strings.Split(moduleName, "/")
+
+	if len(nameParts) < 3 {
```

Comment:
We might want to add a TODO comment here for the future if we find that `dep.Path` cases don't always conform to `host/org/repo` I think there might be some core packages that are just `domain/thing`, but I can't find them currently. Does anyone else have more concrete information on `dep.Path`variations?

Reply:
Good call @spiffcs -- I think this could actually be simplified to just use the `moduleName` directly in the path, since we're using a resolver now